code/modules/food_and_drinks/restaurant/generic_venues.dm

```diff
@@ -39,9 +40,12 @@
 
 /datum/venue/restaurant/on_get_order(mob/living/simple_animal/robot_customer/customer_pawn, obj/item/order_item)
 	. = ..()
-	customer_pawn.visible_message("<span class='danger'>[customer_pawn] pushes [order_item] into their mouth-shaped hole!</span>", "<span class='danger'>You push [order_item] into your mouth-shaped hole.</span>")
+	var/obj/item/food/ordered_food = order_item
+	customer_pawn.visible_message("<span class='danger'>[customer_pawn] pushes [ordered_food] into their mouth-shaped hole!</span>", "<span class='danger'>You push [ordered_food] into your mouth-shaped hole.</span>")
 	playsound(get_turf(customer_pawn),'sound/items/eatfood.ogg', rand(10,50), TRUE)
-	qdel(order_item)
+	total_income += ordered_food.venue_value
```

Comment:
Don't do this, venue_price is a generic element for a reason (and it helps to not rely on /obj/item/food for the soon moth-bots).

Fire a signal that the venue_value element hooks onto.